### PR TITLE
Updating paths in systemctl files and automatically creating systemctl files through CMake

### PR DESCRIPTION
**Pull request by magni-mar**

### CODE REVIEW

File: libs/systemctl/systemd/template.service.in [hunk 1]
```diff
@@ -1,10 +1,10 @@
 [Unit]
```
Comment (jbbjarnason) [inline libs/systemctl/systemd/template.service.in:1]: image


https://fedoramagazine.org/systemd-template-unit-files/

File: exes/ethercat/CMakeLists.txt [hunk 2]
```diff
@@ -66,3 +59,6 @@ install(
     ${CMAKE_INSTALL_BINDIR}/debug/
   CONFIGURATIONS Debug
 )
+
+include("${CMAKE_SOURCE_DIR}/libs/systemctl/CMakeLists.txt")
```
Comment (jbbjarnason) [inline exes/ethercat/CMakeLists.txt:63]: bad practice

File: libs/systemctl/CMakeLists.txt
Comment (jbbjarnason) [inline libs/systemctl/CMakeLists.txt:1]: move to cmake file within cmake folder, example `cmake/tfc_systemd.cmake`

File: libs/systemctl/systemd/template.service.in (hunk 1 above)
Comment (jbbjarnason) [reply libs/systemctl/systemd/template.service.in:1]: `sudo systemctl start ipc-ruler.service`
`sudo systemctl start ethercat.service`
`sudo systemctl start sensor_control@id.service`

File: exes/ethercat/CMakeLists.txt (hunk 2 above)
Comment (magni-mar) [reply exes/ethercat/CMakeLists.txt:63]: moved to top level cmake file

File: libs/systemctl/systemd/template.service.in (hunk 1 above)
Comment (magni-mar) [reply libs/systemctl/systemd/template.service.in:1]: rewritten so that service files are called

`tfc@ethercat.service`
`tfc@ipc-ruler.service`
etc...

Comment (jbbjarnason) [reply libs/systemctl/systemd/template.service.in]: How can we edit id and or log level?

Comment (jbbjarnason) [reply libs/systemctl/systemd/template.service.in]: have you tested if this works like it is now?

Comment (magni-mar) [reply libs/systemctl/systemd/template.service.in]: > How can we edit id and or log level?

one service file is created for each respective executable

Comment (magni-mar) [reply libs/systemctl/systemd/template.service.in]: > How can we edit id and or log level?

do you want to set the log level while the service file is running or when it is created?

Comment (magni-mar) [reply libs/systemctl/systemd/template.service.in]: > have you tested if this works like it is now?

I have ran the cmake command to create the .deb package, the service files get created and they run correctly when I try them on the raspberry pi

Comment (magni-mar) [reply libs/systemctl/systemd/template.service.in]: https://www.freedesktop.org/software/systemd/man/systemd.service.html, under "Service Templates"

systemd service files can take a single argument as template arguments but not more than one. I think a good solution is to create a template file for each log-level where the id is the only template parameter. That is the only variable that is infinitely redefineable.

**Pull request closed without merging**